### PR TITLE
Don't make VBoxDivider inherit from HBoxDivider.

### PR DESCRIPTION
**Pull request by anntzer**
They share some common calculation helpers, but a VBoxDivider is not a
specialization of an HBoxDivider, so just move the helpers out to free
functions (they're staticmethods or nearly so anyways).

Also rename "equivalent" to "equal" and "appended" to "summed", which I
think actually express the constraints better, and are shorter too.

Also document their usage, and PEP8-ify a couple of variables.

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Comment by jklymak** [comment]
While you are fixing all this I don't think any of it works properly with subfigures because it depends on figsize.

**Comment by anntzer** [comment]
Looks like this may be simple enough to fix by replacing `get_size_inches()` by `self._fig.bbox.{width,height} / self._fig.dpi`.  Open a separate issue to track this and I'll put in something on top of this PR and #20072?

**Comment by timhoffm** [comment]
I'm not particularly happy with the style of this. These methods are quite obscure. As private static methods, at least their scope was clear. I find the prefix `_hvboxdiv` also more confusing than helpful: merging both names to hv and abbreviating div makes this too different from `HBoxDivider` to infer a connection.

Suggested minimal improvement: Remove the `_hvboxdiv` prefix and add a comment "Helper function used in HBoxDivider/VBoxDivider".

For future PRs: Please split moving code and renaming variables into separate commits. This is really hard to follow.

**Comment by anntzer** [comment]
Added some more comments, and split the variable renaming into a second commit.  The first one is solely about moving the staticmethods out, now.

The alternative would be to have a private common base class to host the staticmethods, but private base classes come with their own problems...

**Review by timhoffm — APPROVED** [review]
Much simpler to review. Thanks for cleaning up. :smile: 

> The alternative would be to have a private common base class to host the staticmethods, but private base classes come with their own problems...

I intentionally did not suggest that :stuck_out_tongue_winking_eye:

**Comment by anntzer** [comment]
Actually, I further amended the second commit to always put the width first and height second in the internal APIs (instead of having x/y but then height/width).

**Review by jklymak — APPROVED** [review]
